### ISSUE TITLE
Unstable formatting for logical operators

### ISSUE THREAD

**Issue by mlrawlings** [prettier/prettier#7544]
Formatting for logical operator chains with parenthesis are not idempotent.

**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBnCBbOAZaBzANQEMAnAFQAsiYB1ASwBsGAhEuIgawAIAyHrgBTosuKIVKVq9Jq3bc+XYTnzFyVWoxZtOASgDcAHShGlo8WqmbZnXjyNdFmZWNWSNM7fP5KASuyYAnq7q0lpyhsZQpioSIVaeto4iMRbuYTYK0S6xlh7hIAA0IBAADjB00KjIoKQkEADuAAqkCFUoRABuEHQAJoUgAEYkRGAccDAAyiUjdGLIMCQArnBFFDAYDDQUdPCo02BwE607dB07AcjgqFVFs6hwJDCNw3gYRMgAZkQM90UAVqgAB6sEZjSZEESzOCfb6-EAAwETWZ4BhwACKiwg8BhPxWIGmJHuJEuAyIAzgDH6JRIsw0PRgFGQAA4AAxFakQe40YYlS7UuBEjrQooAR0x8GepTaICIqAAtFA4HAesr+mwxXQ2M8iK93kgvriivcMHR5ks8ahkaiMVjofrYXiYGT6PTGUgAExFBZERjIgDCmDelyg0GFIEW9zIZLaBrhHWWAEkoCrYBMwDSygBBZMTGABVE4+4AXyLQA)
```sh
--parser babel
```

**Input:**
```jsx
someLongVarThatWillBreak && (someLongVarThatWillBreak && someLongVarThatWillBreak);
```

**First Output:**
```jsx
someLongVarThatWillBreak &&
  someLongVarThatWillBreak && someLongVarThatWillBreak;
```

**Second Output:**
```jsx
someLongVarThatWillBreak &&
  someLongVarThatWillBreak &&
  someLongVarThatWillBreak;
```

**Expected behavior:**

Print the Second Output for the original Input.

**Comment by evilebottnawi** [comment]
@thorn0 We fix it in the next version, right?

**Comment by thorn0** [comment]
Right. Has has been fixed by #7026.
